src/solver/infeasible-problem-analysis/watched-constraints.cpp

```diff
@@ -159,17 +164,22 @@ std::string HydroProduction::infeasibilityCause()
 ConstraintsFactory::ConstraintsFactory()
 {
     regex_to_ctypes_ = {
-      {"::hourly::", std::make_unique<HourlyBC, const std::string&, const double>},
-      {"::daily::", std::make_unique<DailyBC, const std::string&, const double>},
-      {"::weekly::", std::make_unique<WeeklyBC, const std::string&, const double>},
-      {"^FictiveLoads::", std::make_unique<FictitiousLoad, const std::string&, const double>},
-      {"^AreaHydroLevel::", std::make_unique<HydroLevel, const std::string&, const double>},
-      {"^Level::", std::make_unique<STS, const std::string&, const double>},
-      {"^HydroPower::", std::make_unique<HydroProduction, const std::string&, const double>}};
+      {"::hourly::", std::make_unique<HourlyBC, const std::string&, double>},
+      {"::daily::", std::make_unique<DailyBC, const std::string&, double>},
+      {"::weekly::", std::make_unique<WeeklyBC, const std::string&, double>},
+      {"^FictiveLoads::", std::make_unique<FictitiousLoad, const std::string&, double>},
+      {"^AreaHydroLevel::", std::make_unique<HydroLevel, const std::string&, double>},
+      {"^Level::", std::make_unique<STS, const std::string&, double>},
+      {"^HydroPower::", std::make_unique<HydroProduction, const std::string&, double>},
+      {"^WithdrawalSum::", std::make_unique<STSWithdrawalSum, const std::string&, double>},
+      {"^InjectionSum::", std::make_unique<STSInjectionSum, const std::string&, double>},
+      {"^NettingSum::",
+       std::make_unique<STSNettingSum, const std::string&, double>}}; // namespace
+                                                                      // Antares::Optimization
 }
 
 std::unique_ptr<WatchedConstraint> ConstraintsFactory::create(const std::string& name,
-                                                              const double value) const
+                                                              double value) const
```

Comment:
Here you removed a **const**. Is it on purpose ?

Reply:
yes, `double` are passed by copy. It doesn't really make sense to specify `const`, it means that the copy used inside the function will be const, which is an implementation detail

Reply:
ok, it's just that we make it clear that the parameters will not be changed (whether it's a copy or not)